regolith/broker.py

```diff
@@ -0,0 +1,88 @@
+"""API for accessing the metadata and file storage"""
+from regolith.database import dump_database, open_dbs
+from regolith.runcontrol import DEFAULT_RC, load_rcfile, filter_databases
+from regolith.storage import store_client, push
+
+
+def load_db(rc_file="regolithrc.json"):
+    """Create a Broker instance from an rc file"""
+    rc = DEFAULT_RC
+    rc._update(load_rcfile(rc_file))
+    filter_databases(rc)
+    return Broker(rc)
+
+
+class Broker:
+    """Interface to the database and file storage systems
+
+    Examples
+    --------
+
+    >>> # Load the db
+    >>> db = Broker.from_rc()
+    >>> # Get a docment from the broker
+    >>> ergs =db['group']['ergs']
+    >>> # Store a file
+    >>> db.add_file(ergs, 'myfile', '/path/to/file/hello.txt')
+    >>> # Get a file from the store
+    >>> path = db.get_file(ergs, 'myfile')
+    """
+    def __init__(self, rc=DEFAULT_RC):
+        self.rc = rc
+        # TODO: Lazy load these
+        with store_client(rc) as sclient:
+            self.store = sclient
+        rc.client = open_dbs(rc)
+        self._dbs = rc.client.dbs
+        self.md = rc.client.chained_db
+        self.db_client = rc.client
+
+    def add_file(self, document, name, filepath):
+        """Add a file to a document in a collection.
+
+        Parameters
+        ----------
+        document : dict
+            The document to add the file to
+        name : str
+            Name of the reference to the file
+        filepath : str
+            Location of the file on local disk
+        """
+        output_path = self.store.copydoc(filepath)
```

Comment:
This may need a little change so that the filename is unique. Otherwise we may run into collisions (not all files can be called `fig1.pdf`).

Reply:
I'm going to let this be for now, we currently error on same names without force and that might be ok for now.